File: software/model-benchmark/[Deprecated] wifi-sound-thing-gpt-5-3-codex/android/app/src/test/java/com/wifisound/ui/audio/AudioPacketCodecTest.java
package com.wifisound.ui.audio;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

import org.junit.Test;

public class AudioPacketCodecTest {
    @Test
    public void encodeDecode_roundTrip() {
        byte[] payload = new byte[] { 1, 2, 3, 4, 5, 6 };

        byte[] packet = AudioPacketCodec.encode(
            12,
            456_000L,
            48_000,
            2,
            960,
            payload,
            payload.length
        );

        AudioPacketCodec.DecodedPacket decoded = AudioPacketCodec.decode(packet, packet.length);
        assertNotNull(decoded);
        assertEquals(12, decoded.sequence);
        assertEquals(456_000L, decoded.timestampUs);
        assertEquals(48_000, decoded.sampleRate);
        assertEquals(2, decoded.channels);
        assertEquals(960, decoded.frameSize);
        assertArrayEquals(payload, decoded.payload);
    }

    @Test
    public void decode_rejectsInvalidMagic() {
        byte[] payload = new byte[] { 10, 20, 30, 40 };
        byte[] packet = AudioPacketCodec.encode(
            1,
            1_000L,
            44_100,
            1,
            1024,
            payload,
            payload.length
        );

        packet[0] = 0x00;
        AudioPacketCodec.DecodedPacket decoded = AudioPacketCodec.decode(packet, packet.length);
        assertNull(decoded);
    }

    @Test
    public void decode_rejectsTruncatedPayload() {
        byte[] payload = new byte[] { 7, 8, 9, 10, 11, 12 };
        byte[] packet = AudioPacketCodec.encode(
            9,
            99_000L,
            48_000,
            2,
            480,
            payload,
            payload.length
        );

        AudioPacketCodec.DecodedPacket decoded = AudioPacketCodec.decode(packet, packet.length - 2);
        assertNull(decoded);
    }
}

File: software/model-benchmark/wifi-sound-thing-claude-fable-5/app/src/test/java/app/wifisoundthing/core/AudioPacketCodecTest.kt
package app.wifisoundthing.core

import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class AudioPacketCodecTest {

    @Test
    fun `encode then decode round-trips all fields`() {
        val payload = ByteArray(321) { (it * 7).toByte() }
        val bytes = AudioPacketCodec.encode(seq = 12345L, ptsUs = 987_654_321L, payload = payload)
        val packet = AudioPacketCodec.decode(bytes)!!
        assertEquals(12345L, packet.seq)
        assertEquals(987_654_321L, packet.ptsUs)
        assertArrayEquals(payload, packet.payload)
    }

    @Test
    fun `sequence numbers survive the unsigned 32-bit boundary`() {
        val bigSeq = 0xFFFF_FFF0L
        val packet = AudioPacketCodec.decode(AudioPacketCodec.encode(bigSeq, 0L, byteArrayOf(1)))!!
        assertEquals(bigSeq, packet.seq)
    }

    @Test
    fun `empty payload is valid`() {
        val packet = AudioPacketCodec.decode(AudioPacketCodec.encode(1L, 2L, ByteArray(0)))!!
        assertEquals(0, packet.payload.size)
    }

    @Test
    fun `decode respects explicit length shorter than the array`() {
        val payload = ByteArray(100) { it.toByte() }
        val bytes = AudioPacketCodec.encode(7L, 8L, payload)
        val padded = bytes + ByteArray(50) // simulate a reused datagram buffer
        val packet = AudioPacketCodec.decode(padded, bytes.size)!!
        assertArrayEquals(payload, packet.payload)
    }

    @Test
    fun `garbage and truncated datagrams are rejected, not thrown`() {
        assertNull(AudioPacketCodec.decode(ByteArray(0)))
        assertNull(AudioPacketCodec.decode(ByteArray(5)))
        assertNull(AudioPacketCodec.decode(ByteArray(64) { 0x42 })) // wrong magic
        val wrongVersion = AudioPacketCodec.encode(1L, 1L, byteArrayOf(1)).also { it[2] = 99 }
        assertNull(AudioPacketCodec.decode(wrongVersion))
        val wrongType = AudioPacketCodec.encode(1L, 1L, byteArrayOf(1)).also { it[3] = 9 }
        assertNull(AudioPacketCodec.decode(wrongType))
    }

    @Test(expected = IllegalArgumentException::class)
    fun `oversized payload is refused at encode time`() {
        AudioPacketCodec.encode(1L, 1L, ByteArray(Protocol.MAX_AUDIO_PAYLOAD + 1))
    }
}
